coding/logging.rst

```diff
@@ -0,0 +1,231 @@
+#######
+Logging
+#######
+
+This page provides guidance to developers for using logging with :lmod:`lsst.log`  in the science pipeline code base. 
+Developers are encouraged to insert log messages whenever and wherever they might be useful, with appropriate component names and levels.
+
+.. _logger-names:
+
+Logger Names
+============
+Logger names should generally start with the fully qualified name of the module/file containing the logger, without the leading ``lsst.``.
+If the logger is saved as a variable in a class, it is often appropriate to name the logger after the class.
+Logger names use ``.`` as component separators, not ``::``, even in C++.
+Some example logger names are ``afw.image.MaskedImage``, ``meas.algorithms.starSelector``, etc.
```

Comment:
Actually, I'd recommend moving the `meas.algorithms.starSelector` example as a sentence right after the first "Logger names should generally..." sentence.

Then the note about class and class-based names would be a second paragraph (and the `afw.image.MaskedImage` would be a second sentence of the second paragraph).

The sentence about separators can be on its own as a third paragraph.